ansible_ai_connect_admin_portal/config/webpack.mock.api.keys.modelids.ts

```diff
@@ -0,0 +1,134 @@
+import {
+  delay,
+  DELAY_MS,
+  ERROR_DESCRIPTION,
+  ORG_ID,
+} from "./webpack.mock.globals";
+
+const webpackMockServer = require("webpack-mock-server");
+
+export default webpackMockServer.add((app, helper) => {
+  const KEY_FIELD = "key";
+  const MODEL_ID_FIELD = "model_id";
+  const LAST_UPDATED_FIELD = "last_update";
+
+  const keys = new Map<string, { key: string; last_update: Date }>();
+  const modelIds = new Map<string, { model_id: string; last_update: Date }>();
+
+  //API KEY RELATED ENDPOINTS
+  app.get("/api/v0/wca/apikey/", async (_req, res) => {
+    await delay(DELAY_MS);
+    if (keys.has(ORG_ID)) {
+      // Key response only contains the Last Updated field
+      const data = keys.get(ORG_ID);
+      res.json({ last_update: data[LAST_UPDATED_FIELD] });
+    } else {
+      res.json({});
+    }
+  });
+
+  app.get("/api/v0/wca/apikey/test/", async (_req, res) => {
+    await delay(DELAY_MS);
+    if (!keys.has(ORG_ID)) {
+      res.sendStatus(404);
+      return;
+    } else {
+      const data = keys.get(ORG_ID);
+      const key = data[KEY_FIELD];
+      if (key === "error-test") {
+        // Emulate a server-side error
+        res.status(500).json({ detail: ERROR_DESCRIPTION });
+        return;
+      }
+      if (key === "invalid-test") {
+        // Emulate an invalid API Key
+        res.sendStatus(400);
+        return;
+      }
+    }
+    res.sendStatus(200);
+  });
+
+  app.post("/api/v0/wca/apikey/", async (_req, res) => {
+    await delay(DELAY_MS);
+    const key = _req.body["key"];
+    if (key === "error") {
+      // Emulate a server-side error
+      res.status(500).json({ detail: ERROR_DESCRIPTION });
+      return;
+    }
+    if (key === "invalid") {
+      // Emulate an invalid API Key
+      res.sendStatus(400);
+      return;
+    }
+    const entry = { key: key, last_update: new Date() };
+    keys.set(ORG_ID, entry);
+    res.sendStatus(200);
+  });
+
+  app.delete("/api/v0/wca/apikey/", async (_req, res) => {
+    await delay(DELAY_MS);
+    if (keys.has(ORG_ID)) {
+      keys.delete(ORG_ID);
+      if (modelIds.has(ORG_ID)) {
+        modelIds.delete(ORG_ID);
+      } else {
+        res.sendStatus(404);
```

Comment:
It's plausible the User set an API Key but not the Model ID.

Therefore when deleting, if a Model ID is not found it's probably better to return a 200.
![image](https://github.com/user-attachments/assets/0174876e-0177-47d9-875b-654a68821866)

Reply:
This code also throws:
```
/home/manstis/workspaces/git/manstis/forks/ansible-ai-connect-service/ansible_ai_connect_admin_portal/scripts/start.js:11
  throw err;
  ^

Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at ServerResponse.setHeader (node:_http_outgoing:703:11)
    at ServerResponse.header (/home/manstis/workspaces/git/manstis/forks/ansible-ai-connect-service/ansible_ai_connect_admin_portal/node_modules/express/lib/response.js:794:10)
    at ServerResponse.contentType (/home/manstis/workspaces/git/manstis/forks/ansible-ai-connect-service/ansible_ai_connect_admin_portal/node_modules/express/lib/response.js:624:15)
    at ServerResponse.sendStatus (/home/manstis/workspaces/git/manstis/forks/ansible-ai-connect-service/ansible_ai_connect_admin_portal/node_modules/express/lib/response.js:373:8)
    at /tmp/webpack-mock-server/1728644664039/config/webpack.mock.api.keys.modelids.js:72:17 {
  code: 'ERR_HTTP_HEADERS_SENT'
}
```
This is because you're calling [this](https://github.com/ansible/ansible-ai-connect-service/pull/1333/files#diff-e3431d5d1488e88248bf0b5c69bce36bcad44398ab5439539101cf4cffb55642R77) then [this](https://github.com/ansible/ansible-ai-connect-service/pull/1333/files#diff-e3431d5d1488e88248bf0b5c69bce36bcad44398ab5439539101cf4cffb55642R79).